spec/models/user_spec.rb

```diff
@@ -1,5 +1,27 @@
 require "rails_helper"
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe "validations" do
```

Comment:
In this case, we can use `describe #valid?` because as usual in describe we put some method names or class names. It's good practice
```suggestion
  describe "#valid?" do
```

From by blog:
![image](https://user-images.githubusercontent.com/33006817/128320677-b474e96a-070e-4bba-b8fd-285f80769ac4.png)

Reply:
**Method** `context`
Often this method is confused with the describe method. The context method similarly collects all sorts of checks, but its difference is that this method is used to describe the state of a scenario, rather than a class or method. This method takes a string value or a block as input. Context, according to conventions, begin with the words when, with, given. 

```ruby
describe User do
  let(:user) { build :user, firstname: firstname, lastname: lastname, email: email, password: password }

  let(:firstname) { "Denis" }
  let(:lastname) { "Zaharov" }
  let(:email) { "denis.zaharov@flatstack.com" }
  let(:password) { "password" }

  describe "#valid?" do
    it { expect(user.valid?).to be_truthy }

    context "when firstname is blank" do
      let(:firstname) { nil }

      let(:expected_errors) { "some errors" }
      
      it "validates user" do
        expect(user.valid?).to be_falsey
        expect(user.errors.full_messages).to eq(expected_errors)
      end
    end

    # etc
  end
end
```

Reply:
please, read https://medium.flatstack.com/applying-behavior-driven-development-to-ruby-on-rails-web-applications-using-rspec-cucumber-and-2c893c3ab7eb this article